src/pages/searchPage/searchPageView.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState, useEffect, useCallback } from 'react';
+import { fetchSearchPage, getAllPlanets } from '../../shared/API';
+import { useParams, useNavigate } from 'react-router-dom'; // Import useNavigate
+import SearchBlock from '../../features/SearchBlock/SearchBlockView';
+import ResultsBlock from '../../features/ResultsBlock/ResultsBlockView';
+import ErrorButton from '../../features/ErrorButton/ErrorButtonView';
+
+const SearchPage = () => {
+  const [searchTerm, setSearchTerm] = useState(
+    localStorage.getItem('searchRequest') || ''
+  );
+  const countPerPage = 10;
+  const [countResults, setCountResults] = useState(10);
+  const [searchResults, setSearchResults] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const { page } = useParams();
+  const [currentPage, setCurrentPage] = useState(Number(page) || 1);
+  const navigate = useNavigate();
+
+  const handleSearch = useCallback(
+    async (searchText: string) => {
+      setLoading(true);
+      try {
+        if (searchText === searchTerm) {
+          const searchResults = await fetchSearchPage(searchText, currentPage);
+          setCountResults(searchResults.count);
+          setSearchResults(searchResults.results);
+          setSearchTerm(searchText);
+          localStorage.setItem('searchRequest', searchText);
+        } else {
+          const searchResults = await fetchSearchPage(searchText, 1);
+          setCountResults(searchResults.count);
+          setSearchResults(searchResults.results);
+          setSearchTerm(searchText);
+          localStorage.setItem('searchRequest', searchText);
+          setCurrentPage(1);
+          navigate(`/search/1`);
+        }
+      } catch (error) {
+        setSearchResults([]);
+      } finally {
+        setLoading(false);
+      }
+    },
+    [navigate, searchTerm, currentPage]
+  );
+
+  useEffect(() => {
+    if (page) {
+      setCurrentPage(parseInt(page, 10));
+    }
+  }, [page]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      if (searchTerm) {
+        handleSearch(searchTerm);
+      } else {
+        setLoading(true);
+        const searchResults = await getAllPlanets();
+        setSearchResults(searchResults.results);
+        setSearchTerm('');
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, [searchTerm, currentPage, handleSearch, navigate]);
```

Comment:
looks like`currentPage` and `navigate` are not used inside `useEffect`

Reply:
yes, you right